### PR TITLE
Implemented Strategy pattern to improve readability on operator_install.go file

### LINKED ISSUE #75: Improve readability on operator_install.go file
Break operatorInstall function into modular single purpose functions to improve readability and call those in that central function. One suggestion is to use the strategy pattern where each install mode can be a strategy and be dynamically selected based on its type.

### PR DESCRIPTION
**Pull request by spabba17**
Fixes #75 
Signed-off-by: Soundharya Pabba <spabba@redhat.com>

### CODE REVIEW

**Comment by acmenezes** [comment]
We're working on issue #97 that will change this a bit and then we'll come back here and rebase it and review it.

File: internal/capability/operator_install.go [hunk 1]
```diff
@@ -24,6 +24,18 @@ type operatorData struct {
...
+
+type InstallModeTypeOwnNamespace struct {}
+
+type InstallModeTypeSingleNamespace struct {}
+
+type InstallModeTypeMultiNamespace struct {}
+
+type installModeData interface {
+	createFromOpertorData(data operatorData) (operator.OperatorGroupData)
+ }
```
Comment (skattoju) [inline internal/capability/operator_install.go:37]: would move this to a another file called install_modes.go or something along those lines..

File: internal/capability/operator_install.go [hunk 2]
```diff
@@ -107,45 +119,64 @@ func createGroupByInstallMode(s operator.SubscriptionData, c operator.Client, m
...
+ 
+ func (m *InstallModeTypeSingleNamespace) createFromOperatorData(data operatorData) (operator.OperatorGroupData){
+	 operator.CreateNamespace(context.Background(), data.targetNs1)
+	 opGroupData := operator.OperatorGroupData{
+		 Name:			  data.operatorGroup,
+		 TargetNamespaces: []string{data.targetNs1},
+	 }
+	 return opGroupData
+ }
```
Comment (skattoju) [inline internal/capability/operator_install.go:179]: would move this to a another file called install_modes.go or something along those lines..

**Review by acmenezes — COMMENTED** [review]
A few points to highlight here:

First of all thanks a lot @spabba17 and @makon57 from contributing in such a complicated issue! @spabba17 the idea implemented here seems pretty close to what we were interested in accomplish with this strategy pattern challenge. To that end congratulations! I believe that would be the solution.

The counter-point here is that we still have another issue, more problematic than the requested in this current one. What was that? This portion of code was making too many decisions in a single instance. There is no decoupling of responsibilities and still there are multiple pieces of the project that depends on that information.

The key problematic areas are:
- selecting the proper install mode
- defining targetNamespaces
- creating namespaces
- defining operatorGroup objects
- creating operatorGroup objects
- keeping track of all that information on a wider scope for proper clean up.

Those problems above must be solved separately. Completely decoupled from each other for a few good reasons that are explained in PR https://github.com/opdev/opcap/pull/98

So the above mentioned PR solved all of those problems including the one on this issue. But it opened the door for more sophisticated treatment of install modes and namespace testing. Meaning being able to select install mode from flags, including using all of the supported install modes and integrating with operand install to point out a target namespace different from the own namespace for the operand to be installed. Those new challenges require more collective thought and deserve a separate issue that will replace this one here. Most likely this same pattern applied here can be transported to the new issue when we get it in.

**Comment by acmenezes** [comment]
As state in comment https://github.com/opdev/opcap/pull/95#pullrequestreview-1049865568 we'll close this issue on behalf of opening a new one on install modes when we're ready to adopt new testing strategies for operands.

**Pull request closed without merging**